components/tidb_query/src/codec/mysql/duration.rs

```diff
@@ -646,46 +647,30 @@ impl Duration {
     /// Converts a `Duration` to printable numeric string representation
     #[inline]
     pub fn to_numeric_string(self) -> String {
-        use std::fmt::Write;
-        let mut buf = String::with_capacity(13);
-        if self.neg() {
-            buf.push('-');
-        }
-        write!(
-            buf,
-            "{:02}{:02}{:02}",
-            self.hours(),
-            self.minutes(),
-            self.secs(),
-        )
-        .unwrap();
-        let fsp = self.get_fsp();
-        if fsp > 0 {
-            let nanos = self.subsec_micros() / (TEN_POW[MICRO_WIDTH - usize::from(fsp)]) as u32;
-            write!(buf, ".{:01$}", nanos, fsp as usize).unwrap();
-        }
-        buf
+        self.format("")
     }
 
-    pub fn from_i64(ctx: &mut EvalContext, mut n: i64, fsp: u8) -> Result<Duration> {
-        use crate::codec::error::ERR_TRUNCATE_WRONG_VALUE;
-
+    /// If the error is overflow, the result will be returned, too.
+    /// Otherwise, only one of result or err will be returned
+    pub fn from_i64_without_ctx(mut n: i64, fsp: u8) -> (Option<Duration>, Option<Error>) {
```

Comment:
Seems this function can be a private method.

Reply:
I split out this func from `from_i64` is for the `cast_int_as_duration`, so I should be `pub`